src/util.cpp

```diff
@@ -0,0 +1,43 @@
+#include <regex>
+#include <sstream>
+
+namespace url_filter {
+
+std::string kotlinDecode(const std::string &url) {
```

Comment:
why call it `kotlinDecode`?

Reply:
Didn't come up with a better name 🌚 The function mimics Android/Java's library function's behavior.